source/blog/2016-01-29-dont-run-yum-update-within-a-running-container.html.md

```diff
@@ -0,0 +1,78 @@
+---
+title: Don't run yum update within a running container
+author: baude
+date: 2016-01-29 14:28:54 UTC
+published: false
+comments: true
+---
+
+Recently I have been reviewing a massive collection of Dockerfiles and interacting with customers
+and ISVs alike.  In this time, I have seen all sorts of actions being taken for container design and how
+those containers should be run.  One such action that I really struggle with is when users are either told
+or take the initiative themselves to update packages in a running container.  For example, they execute something like:
+
+```
+docker run -it foobar yum -y update
+
+```
+While there could be a litenany of reasons for doing this,  consider the consequences of doing this:
+
+* Only the container itself is updated, the image remains unchanged.
+* If you were running 10 of these containers, that would result in 10 separate yum updates.
+* With multiple updated containers, perhaps at different times, each container could potentially have different packages in it.
+* The author of the image may not have tested with the updated packages and therefore you may encounter undesirable results.
+
+That said, there are certainly a lot of images on various registries that have grown stale or are not updated with the frequency users desire.  Clearly in this case, it would be ideal if there was an automated build service from the provider that would rebuild the images if its packages were updated.
+
+Absent of that, I think there is a better way.  If we follow the container paradigm, it would be more prudent to update **the image** in question rather than the container.  This can simply be done with docker where you would use the image from the registry as a seed image and then run yum update to create a new image.  The Dockerfile could be as simple as:
+
+```
+# Use the image from the registry as a seed
+FROM 192.168.122.161:5000/foobar
+
+# Run yum update and then clean the cache
+RUN yum -y update && yum clean all
+```
+Then build the image and name it with some obvious marker to differentiate it:
```

Comment:
Wouldn't this be a good case for using tags and `latest` with a consistent local name rather than renaming the image on each update?  This would mean that the Dockerfiles wouldn't need to be modified to use the renamed images on a regular basis, just the consistent local name.